### PR TITLE
Setup to dev/test snapsync with sim architecture

### PR DESCRIPTION
**Pull request by g11tech**
Setup to dev/test snapsync with sim architecture (The aim of this PR is to setup snapsync sim test/dev infra and not really resolve any snap sync issues which are to be addressed seprately)

tl-dr: run sim like this in build repo from packages/client (supply full path to your `DATADIR` in below command)
```bash
NETWORK=mainnet NETWORKID=1337903 ELCLIENT=geth ADD_EOA_STATE=true SNAP_SYNC=true DEBUG_SNAP=client:*  DATADIR=/usr/app/ethereumjs/packages/client/data npm run tape -- test/sim/snapsync.spec.ts
```
and see snap sync sim from geth peer:
image
 

In detail:
FROM: `test/sim/snapsync.md`:

### Snapsync sim setup

1. Start external geth client:
```bash
NETWORK=mainnet NETWORKID=1337903 ELCLIENT=geth  DATADIR=/usr/app/ethereumjs/packages/client/data test
/sim/single-run.sh
```

2. (optional) Add some txs/state to geth
```bash
EXTERNAL_RUN=true ADD_EOA_STATE=true  DATADIR=/usr/app/ethereumjs/packages/client/data npm run tape -- test/sim/snapsync.spec.ts
```

3. Run snap sync:
```bash
EXTERNAL_RUN=true SNAP_SYNC=true  DATADIR=/usr/app/ethereumjs/packages/client/data npm run tape -- test/sim/snapsync.spec.ts
```
you may add `DEBUG_SNAP=client:*` to see client fetcher snap sync debug logs i.e.
```bash
EXTERNAL_RUN=true SNAP_SYNC=true DEBUG_SNAP=client:* DATADIR=/usr/app/ethereumjs/packages/client/data npm run tape -- test/sim/snapsync.spec.ts
```

## Combinations

Combine 2 & 3 in single step:
```bash
EXTERNAL_RUN=true ADD_EOA_STATE=true SNAP_SYNC=true  DATADIR=/usr/app/ethereumjs/packages/client/data npm run tape -- test/sim/snapsync.spec.ts
```

Combine 1, 2, 3 in single step
```bash
NETWORK=mainnet NETWORKID=1337903 ELCLIENT=geth ADD_EOA_STATE=true SNAP_SYNC=true DEBUG_SNAP=client:*  DATADIR=/usr/app/ethereumjs/packages/client/data npm run tape -- test/sim/snapsync.spec.ts
```

### Fully combined scenarios

1. Test syncing genesis state from geth:
```bash
NETWORK=mainnet NETWORKID=1337903 ELCLIENT=geth SNAP_SYNC=true DEBUG_SNAP=client:*  DATADIR=/usr/app/ethereumjs/packages/client/data npm run tape -- test/sim/snapsync.spec.ts
```

2. Add some EOA account states to geth (just add `ADD_EOA_STATE=true` flag to the command)
```bash
NETWORK=mainnet NETWORKID=1337903 ELCLIENT=geth ADD_EOA_STATE=true SNAP_SYNC=true DEBUG_SNAP=client:*  DATADIR=/usr/app/ethereumjs/packages/client/data npm run tape -- test/sim/snapsync.spec.ts
```

3. Add EOA as well as some contract states (just add `ADD_CONTRACT_STATE=true` flag to the command)
TBD

### CODE REVIEW

File: packages/client/lib/sync/snapsync.ts [hunk 1]
```diff
@@ -75,6 +81,28 @@ export class SnapSynchronizer extends Synchronizer {
     return result ? result[1][0] : undefined
   }
 
+  /**
+   * Start synchronizer.
+   * If passed a block, will initialize sync starting from the block.
```
Comment (acolytec3) [inline packages/client/lib/sync/snapsync.ts:86]: What does this comment mean?  There isn't a block parameter in this function.

**Review by scorbajio — DISMISSED** [review]
ACK

**Review by acolytec3 — APPROVED** [review]
LGTM

**Review by acolytec3 — APPROVED** [review]
LGTM

**Comment by holgerd77** [comment]
That's really great work here, thanks a lot, looks fantastic! 🤩 ❤️

**Pull request merged**